### PR TITLE
Add Container runtime---docker install

### PR DESCRIPTION
**Pull request by Stevent-fei**
### Describe what this PR does / why we need it
Install docker:
Modify the docker configuration file through shell commands
Execute the clean script file when uninstalling

### CODE REVIEW

File: pkg/container-runtime/dockerInstaller.go [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+package container_runtime
+
+import (
+	"fmt"
+	"net"
+
+	"github.com/sealerio/sealer/pkg/infradriver"
+)
+
+type ContainerdInstaller struct {
```
Comment (allencloud) [inline pkg/container-runtime/dockerInstaller.go:24]: If this file is named by dockerInstaller.go, then ContainerdInstaller should be located in another file which is named by containerdInstaller.go.

In addition, the file structure should be :
pkg/container-runtime/installer/docker.go 
pkg/container-runtime/installer/containerd.go

File: pkg/container-runtime/installer.go [hunk 2]
```diff
@@ -35,8 +50,21 @@ type Info struct {
...
-//type containerdInstaller struct{}
-type customContainerRuntimeInstaller struct{}
+	if conf.Type == Containerd {
+		containerdInstaller := &ContainerdInstaller{
+			rootfs: driver.GetClusterRootfs(),
+			driver: driver,
+		}
+		return containerdInstaller
+	}
+	return nil
```
Comment (allencloud) [inline pkg/container-runtime/installer.go:69]: this line will cause severe panic. This function cannot return a nil. This function should be defined to return two parameter
```
func NewInstaller(conf Config, driver infradriver.InfraDriver) (Installer, error) {
```
When the input config type is neither docker nor containerd. This function should return an error.

File: pkg/container-runtime/dockerInstaller.go [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+type DockerInstaller struct {
+	Info   Info
+	rootfs string
+	driver infradriver.InfraDriver
+}
+
+func (d *DockerInstaller) InstallOn(hosts []net.IP) (*Info, error) {
+	RemoteChmod := "cd %s  && chmod +x scripts/* && cd scripts && bash init.sh /var/lib/docker %s %s %s %s"
+	info := &Info{
+		Config: Config{
```
Comment (kakaZhou719) [inline pkg/container-runtime/dockerInstaller.go:33]: can be simplified

File: pkg/container-runtime/dockerInstaller.go [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+)
+
+type DockerInstaller struct {
+	Info   Info
+	rootfs string
+	driver infradriver.InfraDriver
+}
+
+func (d *DockerInstaller) InstallOn(hosts []net.IP) (*Info, error) {
+	RemoteChmod := "cd %s  && chmod +x scripts/* && cd scripts && bash init.sh /var/lib/docker %s %s %s %s"
```
Comment (kakaZhou719) [inline pkg/container-runtime/dockerInstaller.go:31]: In my opinion, the parameters of the init.sh script only need to be related to the container runtime installation process. so, `/var/lib/docker`,` DefaultDomain`, `DefaultPort`, Are these parameters unnecessary?

File: pkg/container-runtime/dockerInstaller.go [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+)
+
+type DockerInstaller struct {
+	Info   Info
+	rootfs string
+	driver infradriver.InfraDriver
+}
+
+func (d *DockerInstaller) InstallOn(hosts []net.IP) (*Info, error) {
+	RemoteChmod := "cd %s  && chmod +x scripts/* && cd scripts && bash init.sh /var/lib/docker %s %s"
```
Comment (VinceCui) [inline pkg/container-runtime/dockerInstaller.go:31]: why init.sh?

File: pkg/container-runtime/dockerInstaller.go [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+		err := d.driver.CmdAsync(ip, initCmd)
+		if err != nil {
+			return nil, fmt.Errorf("failed to exec the install docker init command remote: %s", err)
+		}
+	}
+	return info, nil
+}
+
+func (d *DockerInstaller) UnInstallFrom(hosts []net.IP) error {
+	CleanCmd := "cd %s  && chmod +x scripts/* && cd scripts && bash clean.sh"
```
Comment (VinceCui) [inline pkg/container-runtime/dockerInstaller.go:51]: why clean.sh?

File: pkg/container-runtime/dockerInstaller.go (hunk 5 above)
Comment (Stevent-fei) [reply pkg/container-runtime/dockerInstaller.go:31]: init.sh is the script entry for installing docker and k8s, which supports passing parameters to change some configuration files
https://github.com/sealerio/basefs/blob/main/context/rootfs/scripts/init.sh

File: pkg/container-runtime/dockerInstaller.go (hunk 6 above)
Comment (Stevent-fei) [reply pkg/container-runtime/dockerInstaller.go:51]: clean.sh is the script to uninstall docker/containerd

File: pkg/container-runtime/dockerInstaller.go (hunk 5 above)
Comment (VinceCui) [reply pkg/container-runtime/dockerInstaller.go:31]: just use docker.sh

File: pkg/container-runtime/dockerInstaller.go (hunk 6 above)
Comment (VinceCui) [reply pkg/container-runtime/dockerInstaller.go:51]: image

should use docker-uninstall.sh

**Pull request merged**